modules/gapi/test/infer/gapi_infer_onnx_test.cpp

```diff
@@ -305,10 +326,21 @@ class ONNXtest : public ::testing::Test {
         for (size_t i = 0; i < num_out; ++i) {
             const auto info = result[i].GetTensorTypeAndShapeInfo();
             const auto shape = info.GetShape();
-            const auto type = info.GetElementType();
-            cv::Mat mt(std::vector<int>(shape.begin(), shape.end()), toCV(type),
-                       reinterpret_cast<void*>(result[i].GetTensorMutableData<uint8_t*>()));
-            mt.copyTo(outs[i]);
+            const auto type = toCV(info.GetElementType());
+            const std::vector<int> dims(shape.begin(), shape.end());
+            if (type != -1) {
+                cv::Mat(dims, type,
+                        reinterpret_cast<void*>(result[i].GetTensorMutableData<uint8_t*>()))
+                .copyTo(outs[i]);
+            } else {
+                std::vector<int> out_vec;
+                const size_t total = std::accumulate(dims.begin(), dims.end(), 0);
+                const int64_t* ptr = result[i].GetTensorMutableData<int64_t>();
+                for (size_t l = 0; l < total; ++l) {
+                    out_vec.push_back(static_cast<int>(ptr[l]));
+                }
+                cv::Mat(dims, CV_32S, out_vec.data()).copyTo(outs[i]);
```

Comment:
Why don't you write to `outs[i]` immideatly ?

Reply:
Data died in std::vector<cv::Mat>&. It is strange.